### PR TITLE
Deploy prod environment via cloudformation

### PR DESCRIPTION
**Pull request by mcchang**

### CODE REVIEW

**Review by jacknugent — APPROVED** [review]
Approved but I have no idea what's happening here

**Comment by kevincianfarini** [comment]
@mcchang I added a commit for an HTTP to HTTPS redirect. Is this PR good to go now?

File: cloudformation/private-subnet-public-loadbalancer.yml [hunk 1]
```diff
@@ -0,0 +1,181 @@
+AWSTemplateFormatVersion: '2010-09-09'
```
Comment (mcchang) [inline cloudformation/private-subnet-public-loadbalancer.yml:1]: Perhaps add a comment with a reference to the source for this template: https://github.com/nathanpeck/aws-cloudformation-fargate

File: cloudformation/private-subnet-public-loadbalancer.yml [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+        Fn::If:
+          - 'HasCustomRole'
+          - !Ref 'Role'
+          - !Ref "AWS::NoValue"
+      ContainerDefinitions:
+        - Name: !Ref 'ServiceName'
+          Cpu: !Ref 'ContainerCpu'
+          Memory: !Ref 'ContainerMemory'
+          Image: !Ref 'ImageUrl'
+          Environment:
```
Comment (mcchang) [inline cloudformation/private-subnet-public-loadbalancer.yml:81]: nit: can we remove this? I added it to force the container to update

File: cloudformation/public-private-vpc.yml [hunk 3]
```diff
@@ -0,0 +1,536 @@
+AWSTemplateFormatVersion: '2010-09-09'
```
Comment (mcchang) [inline cloudformation/public-private-vpc.yml:1]: same here with adding a ref to the source repo https://github.com/nathanpeck/aws-cloudformation-fargate

File: cloudformation/public-private-vpc.yml [hunk 4]
```diff
@@ -0,0 +1,536 @@
...
+      HealthCheckPath: /
+      HealthCheckProtocol: HTTP
+      HealthCheckTimeoutSeconds: 5
+      HealthyThresholdCount: 2
+      Name: !Join ['-', [!Ref 'AWS::StackName', 'drop-1']]
+      Port: 80
+      Protocol: HTTP
+      UnhealthyThresholdCount: 2
+      VpcId: !Ref 'VPC'
+  # PublicCert:
```
Comment (mcchang) [inline cloudformation/public-private-vpc.yml:305]: can we move this?

**Review by mcchang — COMMENTED** [review]
Looks good! Let's check this in to the repo for now, but I think we need to figure out a cheaper setup.  

The NAT gateways are probably too expensive going forward.  We might be able to get away with using the "Public VPC" stack (: https://github.com/nathanpeck/aws-cloudformation-fargate) for now or paring down the stack we have - it look like we could just run one NAT gateway instead of the redundant 2 that we have.  That's still $0.045/hour => ~$33 a month though.

One constraint we have is this whitelisted IP for accessing some of the APIs.  We might be able to just get way with doing away with all the NAT Gateways and just provisioning an instance with an elastic IP that proxies traffic to SCWC and HCWC.

Costs for November 2021: https://console.aws.amazon.com/billing/home?region=us-west-2#/bills?year=2021&month=11
image

**Comment by kevincianfarini** [comment]
@mcchang awesome. I'm happy to pair on that this week if you'd like.

**Pull request merged**